src/Microsoft.IdentityModel.Abstractions/NullTelemetryClient.cs

```diff
@@ -25,7 +25,9 @@ public class NullTelemetryClient : ITelemetryClient
         /// <remarks>
         /// Private constructor to prevent the default constructor being exposed.
         /// </remarks>
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```

Comment:
Is it the ClientId?

Reply:
Yes, we should never have a null clientId in my opinion. What are you're thoughts on this?

Reply:
What if we do have one?

Reply:
I am not entirely sure what it would happen on the kusto/db side however, not having a clientID for telemetry purposes would basically defeat its purpose I believe.

Reply:
Then in that case add a constructor taking the ClientId?
It's the null telemetry client though. Meaning it probably does nothing?

Reply:
Correct, It doesn't do anything. I think just disabling the alert in here should suffice so I can move on to the other projects. What do you think? @jmprieur

Reply:
We need to be mindful here.

If there is a case that ClientId can be null we should annotate clientId property that it can be null. Otherwise we are not respecting the contract and we defeat the purpose of using the nullable checks. This could results in a NRE.

If the contract should instead be that the clientId cannot be null, we need to change the NullTelemetryClient to always set the property.

Reply:
for example by changing the constructor to take it as a parameter

Reply:
I agree we have to be mindful of this to respect the current contract however, this particular property is tricky because it is not very suitable to use the nullable property as it is written today, changing the signature of the constructor would change its current functionality and we'll have to use some sort of default value in order for that to be viable with is not a good approach in my opinion. I synced with Keegan over this and seems that disabling the alert on this property made the most sense for now.

Reply:
@keegan-caruso do you sign-off on the PR? Thanks for the update @FuPingFranco